File: C++/src/main_menu.cpp
#include "main_menu.hpp"

#include <iostream>
#include <stdexcept>

MainMenu convertIntToMainMenu(int element) {
	element--;
	if (element > static_cast<int>(MainMenu::EXIT) || element < 0) {
		throw std::runtime_error("Некорректный ввод");
	}
	return static_cast<MainMenu>(element);
}

void printMainMenu() {
	for (int i = 0; i < mainMenuElements; i++) {
		std::cout << "[" << i + 1 << "] " << getElementSescription(static_cast<MainMenu>(i))
				  << std::endl;
	}
	std::cout << std::endl;
}

File: Java/MainMenu.java
// Content of src/MainMenu.java
import java.util.InputMismatchException;
import java.util.Scanner;

enum MainMenu {
	CALCULATE_FUNCTION_VALUE("Вычисление значений функции"),
	PRIME_NUMBERS("Тесты на простоту чисел"),
	MATH_SUM("Вычисление ряда"),
	GAME("Увлекательная игра"),
	EXIT("Выход из программы");

	private final String description;

	MainMenu(String description) {
		this.description = description;
	}

	public String getDescription() {
		return description;
	}

	public static int getMainMenuElements() {
		return MainMenu.values().length;
	}

	public static MainMenu convertIntToMainMenu(int element) {
		if (element < 1 || element > MainMenu.values().length) {
			throw new IllegalArgumentException("Некорректный ввод");
		}
		return MainMenu.values()[element - 1];
	}

	public static void printMainMenu() {
		for (int i = 0; i < MainMenu.values().length; i++) {
			System.out.println(
				"[" + (i + 1) + "] " + MainMenu.values()[i].getDescription()
			);
		}
		System.out.println();
	}
}
